add http monitor tests

guyarb opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Draft PRs should be prefixed with `[WIP]` in their title.

-->
### What does this PR do?

- The PR adds a UT for the HTTP monitoring package.
  - The first scenario is slow response from a server with different "agent" configuration for cleaning IDLE connections.
  - The second scenario is requests & responses with actual bodies
  - The third scenario is incomplete requests, which we make sure don't deny us from capturing other requests
- In the PR I had to convert 2 hard coded values into configurable values (`http_map_cleaner_interval_in_s`, `http_idle_connection_ttl_in_s`), for both of them the hard coded values have become the default values.

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succint form, consider
  opening multiple pull requests instead of a single one.
-->

### Motivation

Testing a complex scenario for the eBPF agent.

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

guyarb committed:
network: http: tests: Added configurable read/write timeouts for the test server

guyarb committed:
network: http: tests: Added test for slow response

This test checks the eBPF capturer with various scenarios for receiving a slow response from the test server and multiple IDLE http connection TTL and map cleaner running intervals

guyarb committed:
network: http: Changed the map clenaer interval to a configurable value with 5 minutes as default

guyarb committed:
network: http: tests: Converted the test server into http echo server

The change will allow us to send request body and get the same body in the response. It will allow us to have more scenarios to be tested against.

guyarb committed:
network: http: tests: Checking the eBPF capturing against scenarios in which we had request/response body

We expect to have zero influence from the request/response body on the capturing process, no matter the size of the request/response.
The tests here will examine this thesis against various body sizes

hmahmood commented on pkg/network/http/monitor_test.go:
```diff
@@ -26,12 +31,209 @@ import (
+	fastSrvDoneFn()
+
+	// Ensure all captured transactions get sent to user-space
+	time.Sleep(10 * time.Millisecond)
```
It could be more robust to use `require.Eventually` for this to succeed under more variable conditions (like gitlab).

guyarb replied:
Thanks for the comment!
1. I copied the line from other tests in the same file, should I change it everywhere?
2. Specifically for that test `require.Eventually` won't suffice, as I'm looking for "not included", so I need for every cycle to be "not included"

hmahmood replied:
1. I would leave the other tests for another PR
2. You could do something similar with a loop and a delay for each iteration and if you find the request in any iteration that would be a fail
3. That should not be a problem here right? If any stats are not there after the first iteration, they should be available eventually on another future call to `GetHTTPStats`?

guyarb replied:
1. Agreed
2. I can do it for sure
3. The problem is that I need to ensure a single request (fastReq) did exists

I'll add a commit with a proposed change, lmk what do you think about it

guyarb replied:
Made the change, let me know if it looks good now

hmahmood replied:
This looks good, thanks for making the changes. There are a couple of other spots in the file with the same pattern, can we change those too also? Otherwise lgtm

guyarb replied:
sorry for the late response, was on PTO
i'll go over it and change them as well

guyarb replied:
added to other test as well

guyarb committed:
network: http: tests: Fixed TestHTTPMonitorLoadWithIncompleteBuffers

Reduced number of requests during the load phase to 100.
Closing the channel after all goroutines have ended (otherwise the loop on the channel will be blocked forever)
Changed requestNotIncluded to includesRequest on the 'fast request'

guyarb commented:
Rebased to main

guyarb committed:
network: http: tests: Fixed rebase error

guyarb committed:
network: http: tests: Made TestHTTPMonitorLoadWithIncompleteBuffers more robust

guyarb commented:
rebased

guyarb committed:
network: http: tests: Added more robust check

guyarb committed:
network: http: tests: Fixed flaky tests
